api/pkg/s3/bucketlifecycleput.go

```diff
@@ -0,0 +1,135 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"encoding/xml"
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+	"github.com/micro/go-log"
+
+	"crypto/md5"
+
+	"github.com/opensds/multi-cloud/api/pkg/policy"
+
+	"github.com/opensds/multi-cloud/s3/pkg/model"
+	s3 "github.com/opensds/multi-cloud/s3/proto"
+	"golang.org/x/net/context"
+)
+
+func (s *APIService) BucketLifecyclePut(request *restful.Request, response *restful.Response) {
+	if !policy.Authorize(request, response, "bucket:put") {
+		return
+	}
+	bucketName := request.PathParameter("bucketName")
+	log.Logf("Received request for create bucket: %s", bucketName)
+	ctx := context.Background()
+	bucket, _ := s.s3Client.GetBucket(ctx, &s3.Bucket{Name: bucketName})
+	body := ReadBody(request)
+	log.Logf("Body request is %v\n", body)
+	log.Logf("%x", md5.Sum(body))
+
+	if body != nil {
+		createLifecycleConf := model.LifecycleConfiguration{}
+		log.Logf("Before unmarshal struct is %v\n", createLifecycleConf)
+		err := xml.Unmarshal(body, &createLifecycleConf)
+		log.Logf("After unmarshal struct is %v\n", createLifecycleConf)
+		if err != nil {
+			response.WriteError(http.StatusInternalServerError, err)
+			return
+		} else {
+			s3RulePtrArr := make([]*s3.LifecycleRule, 0)
+			for _, rule := range createLifecycleConf.Rule {
+				s3Rule := s3.LifecycleRule{}
+
+				// assign the fields
+				s3Rule.ID = rule.ID //Assigning the rule ID
+
+				//Assigning the status value to s3 status
+				log.Logf("Status in rule file is %v\n", rule.Status)
+				s3Rule.Status = rule.Status
+
+				//Assigning the filter, using convert function to convert xml struct to s3 struct
+				s3Rule.Filter = convertRuleFilterToS3Filter(rule.Filter)
+
+				// Create the type of transition array
+				s3ActionArr := make([]*s3.Action, 0)
+
+				for _, transition := range rule.Transition {
+
+					//Defining the Transition array and assigning the values tp populate fields
+					s3Transition := s3.Action{Name: "Transition"}
+
+					//Assigning the value of days for transition to happen
+					s3Transition.Days = transition.Days
+
+					//Assigning the backend value to the s3 struct
+					s3Transition.Backend = transition.Backend
+
+					//Assigning the storage class of the object to s3 struct
+					//tier, err := s.class2tier(transition.StorageClass)
+					//if err != nil{
+					//	response.WriteError(http.StatusBadRequest, err)
+					//	return
+					//}
+					//s3Transition.Tier = tier
+
```

Comment:
same here

Reply:
Storage class is lost here, we need to translate storage class to internal tier and put the tier to s3Tansition.Tier.

Reply:
Resolved